core/views.py

```diff
@@ -145,25 +144,82 @@ def create(self, request, *args, **kwargs):
             return Response({"detail": error}, status=status.HTTP_400_BAD_REQUEST)
 
 
-class DataSelectionViewSet(NestedViewSetMixin, viewsets.ModelViewSet):
+class DataSelectionViewSet(viewsets.ModelViewSet):
     serializer_class = DataSelectionSerializer
     queryset = DataSelection.objects.all()
+    lookup_field = "id"
     http_method_names = ["get", "post", "head", "options", "trace"]
 
-    def create(self, request, *args, **kwargs):
+    def create(self, request, *args, upload_id=None, url_id=None):
         serializer = self.get_serializer_class()(data=request.data or request.POST)
-        parent_data = self.get_parents_query_dict()
-        parent_key = next(iter(parent_data))
         if serializer.is_valid():
             ds = DataSelection.objects.create()
             for table in request.data.get("tables", []):
                 tb = Table.objects.create(**table)
                 ds.tables.add(tb)
-            parent_set = getattr(ds, f"{parent_key}_set")
-            parent_set.add(parent_data[parent_key])
+            if upload_id:
+                ds.upload_set.add(upload_id)
+            elif url_id:
+                ds.url_set.add(url_id)
             return Response(self.get_serializer_class()(ds).data, status=status.HTTP_201_CREATED)
 
+    def list(self, request, *args, url_id=None, upload_id=None):
+        if url_id:
```

Comment:
Will this fail if both `url_id` and `upload_id` not provided?

Reply:
It is impossible, this is feature `drf-nested-routers`